dataflowengineoss/src/main/scala/io/joern/dataflowengineoss/queryengine/Engine.scala

```diff
@@ -50,6 +52,9 @@ class Engine(context: EngineContext) {
 
   private val mainResultTable: ResultTable = new ResultTable
 
+  private val started: mutable.Buffer[ReachableByTask] = mutable.Buffer()
+  private var held: List[ReachableByTask]              = List()
```

Comment:
Why is one a mutable Buffer and the other one a immutable List but a var?
Looks like they are used very similar down below.

Reply:
Because I couldn't make up my mind, lol. I'll choose one.